engine/access/state_stream/engine.go

```diff
@@ -141,7 +107,7 @@ func NewEng(
 	e := &Engine{
 		log:                 logger,
 		backend:             backend,
-		server:              server,
+		server:              server.Server(),
```

Comment:
we don't need to hold on to the server reference anymore now that it's started seprately

Reply:
Details [here](https://github.com/onflow/flow-go/pull/4411#discussion_r1223166014)

Reply:
I mean that `e.server` is never called, so it's not necessary to save it on the engine struct